clients/python/girder_client/cli.py

```diff
@@ -64,44 +62,41 @@ def __init__(self, username, password, dryrun, blacklist,
 parser.add_argument('--port', required=False, default=None)
 parser.add_argument('--api-root', required=False, default=None,
                     help='relative path to the Girder REST API')
-parser.add_argument('--blacklist', default='', required=False,
-                    help='comma separated list of filenames to ignore'),
-parser.add_argument('--dryrun', action='store_true',
-                    help='will not write anything to Girder, only report'
-                    ' on what would happen'),
 
+subparsers = parser.add_subparsers(
+    title='subcommands', dest='subcommand', description='Valid subcommands')
+subparsers.required = True
 
-subparsers = parser.add_subparsers(title='subcommands',
-                                   dest='subcommands',
-                                   description='Valid subcommands',)
 _COMMON_OPTIONS = dict(
-    reuse=dict(longname='--reuse', action='store_true',
-               help='use existing items of same name at same location'
-                    ' or create a new one'),
+    reuse=dict(
+        longname='--reuse', action='store_true',
+        help='use existing items of same name at same location or create a new one'),
     leaf_folders_as_items=dict(
-        longname='--leaf-folders-as-items', required=False,
-        action='store_true',
-        help='upload all files in leaf folders'
-             'to a single Item named after the folder'),
+        longname='--leaf-folders-as-items', required=False, action='store_true',
+        help='upload all files in leaf folders to a single Item named after the folder'),
     parent_type=dict(
         longname='--parent-type', required=False, default='folder',
-        help='type of Girder parent target, one of '
-             '(collection, folder, user)'),
+        help='type of Girder parent target, one of (collection, folder, user)'),
+    blacklist=dict(
+        longname='--blacklist', required=False, default='',
+        help='comma-separated list of filenames to ignore'),
+    dryrun=dict(
+        longname='--dryrun', required=False, action='store_true',
```

Comment:
Nit: I'd like this to be named `--dry-run` and the associated variables updated appropriately, e.g. `dryRun`. This would make this option consistent with the others that are longer than one word (e.g. `--parent-type`, `--leaf-folders-as-items`). Moreover, it would make this option consistent with that in other tools (e.g. git).